src/main/java/data/Database.java

```diff
@@ -1,79 +1,19 @@
 package data;
 
-import java.io.FileInputStream;
-import java.io.FileNotFoundException;
-import java.io.FileOutputStream;
+import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.util.ArrayList;
 
-/* This class creates an instance of a database that a file can be written to and read from.
-Assumptions: for each runID, only 1 of each datatype can be associated with it. */
-public class Database {
-  /* Place where files are stored.*/
-  private String directory;
+public interface Database {
+  /* Abstract methods. */
```

Comment:
You can document the intended function of the abstract methods here, on the interface. Then on the implementations of the interface, functions that override the abstract methods only need to document behavior specific to that implementation.